static/app/utils/formatters.tsx

```diff
@@ -147,71 +147,83 @@ export function getDuration(
-      };
-    };
-
-    if (value >= WEEK) {
-      const {quotient, remainder} = divideBy(WEEK);
-      const suffix = abbr ? t('wk') : ` ${tn('week', 'weeks', quotient)}`;
-
-      return `${quotient}${suffix} ${convertDuration(remainder / 1000, abbr)}`;
-    }
-    if (value >= DAY) {
-      const {quotient, remainder} = divideBy(DAY);
-      const suffix = abbr ? t('d') : ` ${tn('day', 'days', quotient)}`;
-
-      return `${quotient}${suffix} ${convertDuration(remainder / 1000, abbr)}`;
-    }
-    if (value >= HOUR) {
-      const {quotient, remainder} = divideBy(HOUR);
-      const suffix = abbr ? t('hr') : ` ${tn('hour', 'hours', quotient)}`;
-
-      return `${quotient}${suffix} ${convertDuration(remainder / 1000, abbr)}`;
-    }
-    if (value >= MINUTE) {
-      const {quotient, remainder} = divideBy(MINUTE);
-      const suffix = abbr ? t('min') : ` ${tn('minute', 'minutes', quotient)}`;
-
-      return `${quotient}${suffix} ${convertDuration(remainder / 1000, abbr)}`;
+export function getExactDuration(
+  seconds: number,
+  abbreviation: boolean = false,
+  minDuration: string = 'milliseconds'
+) {
+  const operation = seconds < 0 ? Math.ceil : Math.floor;
+  const levels: Level[] = [
+    [operation(seconds / SECS_IN_WKS), abbreviation ? 'wk' : ' weeks'],
+    [operation((seconds % SECS_IN_WKS) / SECS_IN_DAYS), abbreviation ? 'd' : ' days'],
+    [
+      operation(((seconds % SECS_IN_WKS) % SECS_IN_DAYS) / SECS_IN_HRS),
+      abbreviation ? 'hr' : ' hours',
+    ],
+    [
+      operation((((seconds % SECS_IN_WKS) % SECS_IN_DAYS) % SECS_IN_HRS) / SECS_IN_MIN),
+      abbreviation ? 'min' : ' minutes',
+    ],
+    [
+      operation((((seconds % SECS_IN_WKS) % SECS_IN_DAYS) % SECS_IN_HRS) % SECS_IN_MIN),
+      abbreviation ? 's' : ' seconds',
+    ],
+    [
+      operation(
+        (((((seconds * MS_IN_S) % (SECS_IN_WKS * MS_IN_S)) % (SECS_IN_DAYS * MS_IN_S)) %
+          (SECS_IN_HRS * MS_IN_S)) %
+          (SECS_IN_MIN * MS_IN_S)) %
+          MS_IN_S
+      ),
+      abbreviation ? 'ms' : ' milliseconds',
+    ],
+  ];
+  let returntext = '';
+
+  for (let i = 0, max = levels.length; i < max; i++) {
+    if (
+      (i === max - 1 || minDuration === levels[i][1].trim()) &&
+      !returntext &&
+      !levels[i][0]
+    ) {
+      returntext = '0' + levels[i][1];
+      break;
     }
-    if (value >= SECOND) {
-      const {quotient, remainder} = divideBy(SECOND);
-      const suffix = abbr ? t('s') : ` ${tn('second', 'seconds', quotient)}`;
-
-      return `${quotient}${suffix} ${convertDuration(remainder / 1000, abbr)}`;
+    if (levels[i][0] === 0) {
+      continue;
     }
-
-    if (value === 0) {
-      return '';
+    const singular = levels[i][1].substring(0, levels[i][1].length - 1);
+    const duration = levels[i][1];
+    const quotient = levels[i][0];
+
+    returntext +=
+      ' ' + quotient + abbreviation
+        ? // eslint-disable-next-line sentry/no-dynamic-translations
```

Comment:
Let me know if that makes sense. Basically when using the translations it's not possible to have entire "dynamic" strings, since now those can't be extracted out to be given to the translators.

This is similar to why when translating it's usually important not to "build" or concatenate strings to form a sentence, since different languages my have different word orderings. For example:

```tsx
const ask = askType = t('can you buy') : t('will you pickup');
const thing = thingType = t('milk') : t('eggs');
const person = t('for me %s', personName);

return `${ask} ${thing} ${person}`
```

which might produce "can you buy milk for me evan"

But in another language the translation of that is

```
「evan」さんは「milk」をかいます   ください
 ^           ^       ^         ^
 person      thing   ask (buy) ask (can you)
```

It would be impossible for the translator to construct that translation because the translation is broken apart

However if it was done as

```
// Actually not 100% sure if it would be OK to have the object 
// of a sentence translated separately
const thing = thingType = t('milk') : t('eggs');

return askType
  ? t('can you buy %s for me %s', thing, personName)
  : t('will you pickup %s for me %s', thing, personName)
```

now that can be translated into other languages with different word order

Reply:
ah nice yea that makes sense.

stepping away from this pr for now, but can come back to it later